### DISCUSSION TITLE
Is there a way to create a general index at end of a book?

### DISCUSSION THREAD

**Discussion by naish** [quarto-dev/quarto-cli#10213] [Q&A] [answered]
### Description

I am wondering if there is a way to create a general index that would appear at the end of a book, similar to [Jupyter Book](https://jupyterbook.org/en/stable/content/content-blocks.html#indexes). Ideally, it renders as a [list of links](https://jupyterbook.org/en/stable/genindex.html) for HTML and includes page numbers when rendered as print document (e.g., LaTeX → PDF).

**Comment by mcanouil** [comment]
You mean a glossary?

- https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=glossary
- https://github.com/quarto-dev/quarto-cli/issues/1697

**Reply by naish** [reply]
No, I'm thinking of a standard index that one would find at the back of a reference book (think a recipe book or textbook). A reader can look up a topic by keyword or subject and it would indicate which pages within the book that particular topic is discussed. In an HTML context, it would provide one or more hyperlinks to the relevant section(s) in the book.

For example: 
![Print index](https://dataschool.com/assets/images/sql-optimization/how_to_index/Index.png)
)

or https://jupyterbook.org/en/stable/genindex.html

**Reply by mcanouil** [reply, marked as answer]
For LaTeX (PDF) format you can use the LaTeX machinery.
This is for now unsupported by Quarto itself as glossary or basically anything like that.

**Reply by putzwasser** [reply]
I'm facing the same issue: I need to create a glossary for a PDF output. From what I read this isn't supported, yet. But as you said, @mcanouil, one can use plain LaTeX.

I used the template partials to test if/how this is possible.

To the file  `before-title.tex` I added:

```latex
\usepackage[toc,acronym]{glossaries}
\makeglossaries

\newglossaryentry{csf}
{
  name={csf},
  description={A critical success factor (CSF) is an element necessary for a project to achieve its output.}
}
```

To the file  `before-bib.tex` I added:

```latex
\glsaddall
\printglossaries
```

Quarto/pandoc runs through without issues. My only problem is that the execution of `makeglossaries` isn't added to the render call stack. As a result, `\printglossaries` won't show anything.

How can I add this to the call stack?

**Reply by mcanouil** [reply]
@putzwasser Could you share a small reproducible example? Likely a Git repository.
